vertical-pod-autoscaler/pkg/updater/inplace/inplace_recommendation_provider.go

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package inplace
+
+import (
+	"fmt"
+
+	core "k8s.io/api/core/v1"
+	"k8s.io/klog/v2"
+
+	"k8s.io/autoscaler/vertical-pod-autoscaler/pkg/admission-controller/resource/pod/recommendation"
+
+	vpa_types "k8s.io/autoscaler/vertical-pod-autoscaler/pkg/apis/autoscaling.k8s.io/v1"
+	"k8s.io/autoscaler/vertical-pod-autoscaler/pkg/utils/limitrange"
+	vpa_api_util "k8s.io/autoscaler/vertical-pod-autoscaler/pkg/utils/vpa"
+)
+
+type inPlaceRecommendationProvider struct {
+	limitsRangeCalculator   limitrange.LimitRangeCalculator
+	recommendationProcessor vpa_api_util.RecommendationProcessor
+}
+
+// NewInPlaceRecommendationProvider constructs the recommendation provider that can be used to determine recommendations for pods.
+func NewInPlaceRecommendationProvider(calculator limitrange.LimitRangeCalculator,
+	recommendationProcessor vpa_api_util.RecommendationProcessor) recommendation.Provider {
+	return &inPlaceRecommendationProvider{
+		limitsRangeCalculator:   calculator,
+		recommendationProcessor: recommendationProcessor,
+	}
+}
+
+// GetContainersResourcesForPod returns recommended request for a given pod.
+// The returned slice corresponds 1-1 to containers in the Pod.
+func (p *inPlaceRecommendationProvider) GetContainersResourcesForPod(pod *core.Pod, vpa *vpa_types.VerticalPodAutoscaler) ([]vpa_api_util.ContainerResources, vpa_api_util.ContainerToAnnotationsMap, error) {
```

Comment:
I don't feel super great about most of this code being copied from the other provider... All I can spot here is that we removed the annotations generated by the other implementation, is there anything else?

If not, let's just use the canonical provider instead?

(I totally can see how this will be useful in the future but maybe we keep it simple for now?)